papersite/catalog.py

```diff
@@ -1,132 +1,365 @@
-                           keyword=keyword)
+                          entry=k,
+                          papers=with_description(papers),
+                          keyword=keyword)
+
 
 @app.route('/author/<string:fullname>', methods=['GET'])
 def author(fullname):
-    a=query_db("select authorid from authors where      \
-                       fullname = ?", 
-                      [fullname], one=True)
+    a = query_db("select authorid from authors where      \
+                       fullname = ?",
+                 [fullname], one=True)
     if (a is None):
-        return render_catalog('catalog/papers-of-author.html', 
-                               error = "Author " + \
-                                        fullname + " is not found",
-                               fullname=fullname)
-    authorid=a['authorid']
-    papers=query_db("select p.*                                \
+        return render_catalog('catalog/papers-of-author.html',
+                              error="Author " +
+                              fullname + " is not found",
+                              fullname=fullname)
+    authorid = a['authorid']
+    papers = query_db("select p.*                                \
                      from papers as p,                         \
                           papers_authors as pa                 \
                      where                                     \
                            p.deleted_at is null and            \
                            p.paperid=pa.paperid and            \
                            pa.authorid = ?                     \
-                     order by p.lastcommentat desc",[authorid])
+                     order by p.lastcommentat desc", [authorid])
     return render_catalog('catalog/papers-of-author.html',
-                           papers=with_description(papers),
-                           fullname=fullname)
+                          entry=a,
+                          papers=with_description(papers),
+                          fullname=fullname)
 
 
 @app.route('/catalog', methods=['GET'])
 def catalog():
     if request.args.get('q'):
         q = '%' + request.args.get('q') + '%'
-        papers=query_db("select * from papers                      \
+        papers = query_db("select * from papers                      \
                          where                                     \
                                deleted_at is null and              \
                                lower(title) like  lower(?)         \
-                         order by title",[q])
-    else: 
+                         order by title", [q])
+    else:
         papers = []
     return render_catalog('catalog/catalog-search.html',
-                           papers=with_description(papers))
+                          papers=with_description(papers))
+
+
+####################### My Modifications ##############################
```

Comment:
I modified this to "modification made by my GitHub username", so it will help me keep track of what I did in order to write the PFE report, but you can delete this line when merging the branches.